02-Generics/Generics/Generics.cs

```diff
@@ -1,174 +1,113 @@
-		///  "1:00:00,0:00:30" for TimeSpan =>  { new TimeSpan(1, 0, 0), new TimeSpan(0, 0, 30) },
-		///  </example>
 		public static IEnumerable<T> ConvertToList<T>(this string list) {
-			// TODO : Implement ConvertToList<T>
-			// HINT : Use TypeConverter.ConvertFromString method to parse string value
-			throw new NotImplementedException();
-		}
+			if (list == null)
+				throw new ArgumentNullException("List is null");
+
+			var result = new List<T>();
+			foreach (var item in list.Split(ListSeparator))
+				result.Add((T)TypeDescriptor.GetConverter(typeof(T)).ConvertFromString(item));
 
+			return result;
+		}
 	}
 
 	public static class ArrayExtentions {
-
-		/// <summary>
-		///   Swaps the one element of source array with another
-		/// </summary>
-		/// <typeparam name="T">required type of</typeparam>
-		/// <param name="array">source array</param>
-		/// <param name="index1">first index</param>
-		/// <param name="index2">second index</param>
 		public static void SwapArrayElements<T>(this T[] array, int index1, int index2) {
-			// TODO : Implement SwapArrayElements<T>
-			throw new NotImplementedException();
-		}
-
-		/// <summary>
-		///   Sorts the tuple array by specified column in ascending or descending order
-		/// </summary>
-		/// <param name="array">source array</param>
-		/// <param name="sortedColumn">index of column</param>
-		/// <param name="ascending">true if ascending order required; otherwise false</param>
-		/// <example>
-		///   source array : 
-		///   { 
-		///     { 1, "a", false },
-		///     { 3, "b", false },
-		///     { 2, "c", true  }
-		///   }
-		///   result of SortTupleArray(array, 0, true) is sort rows by first column in a ascending order: 
-		///   { 
-		///     { 1, "a", false },
-		///     { 2, "c", true  },
-		///     { 3, "b", false }
-		///   }
-		///   result of SortTupleArray(array, 1, false) is sort rows by second column in a descending order: 
-		///   {
-		///     { 2, "c", true  },
-		///     { 3, "b", false }
-		///     { 1, "a", false },
-		///   }
-		/// </example>
-		public static void SortTupleArray<T1, T2, T3>(this Tuple<T1, T2, T3>[] array, int sortedColumn, bool ascending) {
-			// TODO :SortTupleArray<T1, T2, T3>
-			// HINT : Add required constraints to generic types
+			(array[index1], array[index2]) = (array[index2], array[index1]);
 		}
 
+		public static void SortTupleArray<T1, T2, T3>(this Tuple<T1, T2, T3>[] array, int sortedColumn, bool ascending)
+		where T1 : IComparable
+		where T2 : IComparable
+		where T3 : IComparable
+        {
+            if (sortedColumn == 0)
+                Array.Sort(array, (x, y) => y.Item1.CompareTo(x.Item1));
+            else if (sortedColumn == 1)
+                Array.Sort(array, (x, y) => y.Item2.CompareTo(x.Item2));
+            else if (sortedColumn == 2)
+                Array.Sort(array, (x, y) => y.Item3.CompareTo(x.Item3));
+            else
```

Comment:
looks like duplicate logic. Let's think about a more universal solution.

use Array.Sort(array, (x, y) => ....  only once
(if you think for a long time, then come for help)